fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanTransaction.java

```diff
@@ -963,6 +939,18 @@ public BigDecimal getAmount() {
         return amount;
     }
 
+    public boolean isBefore(final LocalDate date) {
+        return DateUtils.isBefore(getTransactionDate(), date);
+    }
+
+    public boolean isAfter(final LocalDate date) {
+        return DateUtils.isAfter(getTransactionDate(), date);
+    }
+
+    public boolean isOn(final LocalDate date) {
```

Comment:
Personnaly i prefer isEqual... :/

Reply:
I think isEqual would be misleading, because the transaction itself is not equal with a date, it was just performed on that date.